app_dart/lib/src/model/firestore/pr_check_runs.dart

```diff
@@ -47,10 +47,20 @@ class PrCheckRuns extends Document {
   }
 
   /// The json string of the pullrequest belonging to this document.
-  PullRequest get pullRequest => PullRequest.fromJson(
-    json.decode(fields![kPullRequestField]!.stringValue!)
-        as Map<String, Object?>,
-  );
+  PullRequest get pullRequest {
+    final jsonData =
+        jsonDecode(fields![kPullRequestField]!.stringValue!)
+            as Map<String, Object?>;
+    final result = PullRequest.fromJson(jsonData);
+
+    // Workaround for https://github.com/flutter/flutter/issues/166022.
+    if (jsonData['labels'] case final List<Object?> labelData) {
```

Comment:
I don't like the external library. It's not maintained (last commit was 11 months ago and we had to do another workaround). Sigh.